src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1933,44 +1935,109 @@ GenTree* Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
     GenTree* tmp2 = nullptr;
     GenTree* tmp3 = nullptr;
 
-    bool   isConstant     = GenTreeVecCon::IsHWIntrinsicCreateConstant(node, simd32Val);
+    bool isConstant = false;
+    if (simdSize != 64)
+    {
+        isConstant = GenTreeVecCon::IsHWIntrinsicCreateConstant(node, simd32Val);
+    }
+    else
+    {
+        isConstant = GenTreeVecCon::IsHWIntrinsicCreateConstant(node, simd64Val);
+    }
```

Comment:
Is this split to handle the lack of true `simd64` constants or for another reason, like JIT throughput?

Reply:
It's  to handle the lack of true `simd64 `constants

Reply:
Should there be a TODO here to track the fix? I believe I saw one in most of the other places.

Reply:
Added TODO